ctapipe/io/datawriter.py

```diff
@@ -616,6 +613,20 @@ def table_name(self, tel_id):
         """construct dataset table names depending on chosen split method"""
         return f"tel_{tel_id:03d}"
 
+    def _write_trigger(self, writer: TableWriter, event: ArrayEventContainer):
+        """
+        Write trigger information
+        """
+        self._writer.write(
+            table_name="dl1/event/subarray/trigger",
+            containers=[event.index, event.trigger],
+        )
+
+        for tel_id, trigger in event.trigger.tel.items():
+            writer.write(
+                "dl1/event/telescope/trigger", (_get_tel_index(event, tel_id), trigger)
+            )
+
```

Comment:
Quick question for my understanding.
`self._write_trigger(self._writer, event)` is called in L304, which means z
`writer=self._writer`.
But `self._writer` is already available inside the the function.
In which case do you want to use a different writer with this function?

Reply:
It could have been because DataWriter was meant to support other types of TableWriter in the future as well (only HDF5 is currently implemented, but it would be pretty easy to have other formats). But even then, we could assume the TableWriter itself supports handling of N tables.

Reply:
This was a copy paste mistake, I moved the subarray writing from __call__ where it directly accessed self._writer

Will change.

Since this is all internal anyway, I could also just remove the wrote argument though